packages/slate-editor/src/modules/nodes-hierarchy/hierarchySchema.ts

```diff
@@ -94,6 +94,14 @@ export const hierarchySchema: NodesHierarchySchema = {
     ],
     [TABLE_NODE_TYPE]: [
         allowChildren(isTableRowNode, combineFixers([unwrapSameTypeChild, liftNodeNoSplit])),
+        (editor, path) =>
+            TablesEditor.isTablesEditor(editor) && TablesNormalization.removeEmptyRow(editor, path),
+        (editor, path) =>
+            TablesEditor.isTablesEditor(editor) &&
+            TablesNormalization.normalizeTableData(editor, path),
+        (editor, path) =>
+            TablesEditor.isTablesEditor(editor) &&
+            TablesNormalization.restoreAbsentCells(editor, path),
```

Comment:
imo i would move all normalizations binding to the editor, and export normalizators from package. In this case we can see all in one place, can disable if we need, change order (put smth in the middle because order matters), override default normalization etc.

Reply:
here for example new normalization can work only after wrong nodes will be unwrapped or lifted

Reply:
> here for example new normalization can work only after wrong nodes will be unwrapped or lifted

I read this as "*the hierarchy normalization should run before extension normalizations*" :)

Reply:
But how can we control the order, if is just override editor.normalize

Reply:
> But how can we control the order, if is just override editor.normalize

I'd say we just put the hierarchy normalization last in this list:

```diff
packages/slate-editor/src/modules/editor/createEditor.ts
@@ -35,7 +35,6 @@
         withReact,
         withHistory,
         withNonEmptyValue,
-        withNodesHierarchy(hierarchySchema),
         withBreaksOnExpandedSelection,
         withBreaksOnVoidNodes,
         withInlineVoid(getExtensions),
@@ -47,5 +46,6 @@
         withSerialization(getExtensions),
         ...overrides,
         ...plugins,
+        withNodesHierarchy(hierarchySchema),
     ])(baseEditor);
 }
```